README.md

```diff
@@ -112,6 +112,16 @@ Top-level options exposed by BunnyHop
 | url | AMQP URL to connect to  | `'amqp://localhost'` |
 | serialization | manager to use for serializing and deserializing messages | `require('./lib/serialization/json')` |
 | connectionManager | manager to use for handling connection to AMQP server | `require('./lib/connectionManager')` |
+| onHandlerCompleted| function with signature is onHandlerCompleted({*} Result, {Array<*>} OriginalHandlerArguments) called when listen/subscribe handler completes.| undefined |
+| onHandlerError| function with signature onHandlerError(Error) called when listen/subscribe throws an error/returns rejected promise | undefined |
+
+**Note**
+
+The `onHandlerCompleted` does not affect return values of the handler. It is purely a side effect.
+
+The `onHandlerError` hooks does not affect error bubbling. It is purely a side effect. 
```

Comment:
@krisread does this make sense? or should it not re-throw the errors?

Reply:
Should they just be named onComplete  / onError? 

Does onHandlerCompleted get called only for success or both success and error?

Should the original msg be passed into these functions when they are called - e.g. if we wanted to log what message we are taking about in the event of an error?

Reply:
onSuccess makes sense. I guess I was thinking `handler completed runnning` cause an error would imply it didn't complete...? I guess your suggestion is more clear.

The original message IS passed in. 2nd argument :)

What I was asking about is whether or not it should prevent bubbling of the error or not. If I enable an `onHandlerError` function, should the app still see an "Unhandled Rejection"

Reply:
The way it is not only ADDS side effects. is it better to actually **intercept** the result or error?